builder/build_validate.go

```diff
@@ -254,46 +263,82 @@ func (b *Builder) mcaPkgInfo(mInfo map[string]*swupd.Manifest, version, download
 	pInfo := make(map[string]*mcaBundlePkgInfo)
 
 	// Download and query file metadata from all packages in each bundle
+	bundleWorker := func() {
+		for m := range mCh {
+			var pkgList = []string{}
+			for pkg := range mInfo[m.name].BundleInfo.AllPackages {
+				pkgList = append(pkgList, pkg)
+			}
+
+			out, err := downloadRpms(packagerCmd, pkgList, buildVersionDir, downloadRetries)
+			if err != nil {
+				errorCh <- err
+				wg.Done()
+				return
+			}
+
+			// Collect metadata to resolve installed RPM file names
+			bundlePkgInfo := pkgInfoFromNoopInstall(out.String())
+			for _, p := range bundlePkgInfo {
+				// Resolve files for package when it doesn't exist in the cache
+				rw.RLock()
+				if pkgCache[p.name] == nil {
+					p.files, err = b.resolvePkgFiles(p, version)
+					if err != nil {
+						rw.RUnlock()
+						errorCh <- err
+						wg.Done()
+						return
+					}
+					m.allPkgs[p.name] = p
+				} else {
+					m.allPkgs[p.name] = pkgCache[p.name]
+				}
+				rw.RUnlock()
+
+				rw.Lock()
+				if pkgCache[p.name] == nil {
+					pkgCache[p.name] = p
+				}
+				rw.Unlock()
```

Comment:
by implementing my suggestion above, this block can move inside the if block above since it wont be inside a lock anymore.